server/pulp/server/webservices/views/search.py

```diff
@@ -154,6 +151,12 @@ def _generate_response(cls, query, options, *args, **kwargs):
         if hasattr(cls, 'model'):
             search_method = cls.model.objects.find_by_criteria
         else:
+            # id field should always be in query.fields. We are forced to do this twice, here and
+            # in the queryset for the mongoengine work. This is necessary because in mongoengine
+            # id is an alias to _id, and queries must be translated to reflect that. If we append
+            # id before this split, our serializers will translate this id also.
+            if query.fields and 'id' not in query.fields:
```

Comment:
Just for entertainment value, another very-slightly-less-wordy way to do this is:

```
if 'id' not in query.fields or []:
```

Definitely no need to change, but I thought the alternative is worth at least knowing about.